### PR TITLE
[MFTF] Check visibility block only x product left for simple product

### LINKED ISSUE #28755: [Issue] [MFTF] Check visibility block only x product left for simple product
This issue is automatically created based on existing pull request: magento/magento2#27549: [MFTF] Check visibility block only x product left for simple product

---------

### Preconditions (*)

This MFTF test checks visibility only x product left block for simple product
Screenshot 2020-04-02 at 16 52 52

### Steps to reproduce (*)

1. Create a Simple product with qty =1000 for ex;
2. Go to admin-> Stores -> Configuration -> Catalog -> Inventory -> Stock Options  and set **Only X left Threshold** to **1000** for example;
3. Go to Storefront and open simple product page;

### Expected Result (*) 

:heavy_check_mark: only x left block of the product should be displayed for a simple product

Screenshot from 2020-06-17 14-30-07

### PR DESCRIPTION
**Pull request by Usik2203**
### Description (*)

This MFTF test checks visibility only x product left block for simple product
Screenshot 2020-04-02 at 16 52 52

### Resolved issues:
1. [x] resolves magento/magento2#28755: [MFTF] Check visibility block only x product left for simple product

### CODE REVIEW

**Comment by rodrigowebjump** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump, here is your [new Magento instance](https://pr-27549.instances.magento-community.engineering).
Admin access: [https://pr-27549.instances.magento-community.engineering/admin_32eb](https://pr-27549.instances.magento-community.engineering/admin_32eb)
Login: `2437fc6e` Password: `0353f67fa15b`
Instance will be terminated in up to 3 hours.

**Comment by Usik2203** [comment]
Hi @rodrigowebjump 
Any news about this PR ?
Could I kindly ask you to check it If you will have free slot of time
Thank you a lot !

**Comment by Usik2203** [comment]
@magento run all tests

**Comment by Usik2203** [comment]
Hi @rogyar Could I kindly ask you to review this PR ?
Thanks.

File: app/code/Magento/Catalog/Test/Mftf/Test/StorefrontOnlyXProductLeftForSimpleProductsTest.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            <description value="See Only * Left on product page if Only X left Threshold was set"/>
+            <severity value="MINOR"/>
+        </annotations>
+        <before>
+            <createData entity="_defaultCategory" stepKey="createCategory"/>
+            <createData entity="ApiSimpleProduct" stepKey="createProduct">
+                <requiredEntity createDataKey="createCategory"/>
+            </createData>
+            <magentoCLI command="config:set {{CatalogInventoryOptionsOnlyXleftThreshold.path}} 10000" stepKey="setStockThresholdQty"/>
+            <magentoCLI command="cache:flush" stepKey="flushCache"/>
```
Comment (rogyar) [inline app/code/Magento/Catalog/Test/Mftf/Test/StorefrontOnlyXProductLeftForSimpleProductsTest.xml]: Could we replace flushing the cache with clearing the config cache? Flushing the cache slows down the testing routine significantly every time it's invoked. 

Thank you.

File: app/code/Magento/Catalog/Test/Mftf/Section/StorefrontProductPageSection.xml [hunk 2]
```diff
@@ -25,5 +25,6 @@
         <element name="customOptionDropDown" type="select" selector="//*[@id='product-options-wrapper']//select[contains(@class, 'product-custom-option admin__control-select')]"/>
         <element name="qtyInputWithProduct" type="input" selector="//tr//strong[contains(.,'{{productName}}')]/../../td[@class='col qty']//input" parameterized="true"/>
         <element name="customOptionRadio" type="input" selector="//span[contains(text(),'{{customOption}}')]/../../input" parameterized="true"/>
+        <element name="onlyProductsLeft" type="block" selector="//div[@class='availability only']"/>
```
Comment (rogyar) [inline app/code/Magento/Catalog/Test/Mftf/Section/StorefrontProductPageSection.xml]: Please, make the selector of this element more sophisticated. Like, include some parent classes as well. There's a chance of a second element on the page with "availability only" class.

Thank you.

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by Usik2203** [comment]
@magento run all tests

File: app/code/Magento/Catalog/Test/Mftf/Test/StorefrontOnlyXProductLeftForSimpleProductsTest.xml (hunk 1 above)
Comment (Usik2203) [reply app/code/Magento/Catalog/Test/Mftf/Test/StorefrontOnlyXProductLeftForSimpleProductsTest.xml]: **Done **

File: app/code/Magento/Catalog/Test/Mftf/Section/StorefrontProductPageSection.xml (hunk 2 above)
Comment (Usik2203) [reply app/code/Magento/Catalog/Test/Mftf/Section/StorefrontProductPageSection.xml]: **Done**

**Comment by Usik2203** [comment]
Hi @rogyar 
I have applied your code review recommendations 
Thanks

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-7682](https://jira.corp.magento.com/browse/ENGCOM-7682) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Alfa** [comment]
@magento create issue

**Comment by engcom-Echo** [comment]
@magento run all tests

**Pull request merged**